// vant-green@1.0.44/es/index.js
// This file is auto generated by build/build-entry.js
import ActionSheet from './action-sheet';
import AddressEdit from './address-edit';
import AddressList from './address-list';
import Area from './area';
import Avatar from './avatar';
import BankCard from './bank-card';
import BasicLayout from './basic-layout';
import Body from './body';
import Button from './button';
import Call from './call';
import Card from './card';
import CascadeSelect from './cascade-select';
import Cell from './cell';
import CellGroup from './cell-group';
import Checkbox from './checkbox';
import CheckboxGroup from './checkbox-group';
import Circle from './circle';
import Col from './col';
import Collapse from './collapse';
import CollapseItem from './collapse-item';
import Color from './color';
import ContactCard from './contact-card';
import ContactEdit from './contact-edit';
import ContactList from './contact-list';
import CountDown from './count-down';
import Coupon from './coupon';
import CouponCell from './coupon-cell';
import CouponList from './coupon-list';
import DatetimePicker from './datetime-picker';
import Dialog from './dialog';
import Divider from './divider';
import DoctorCard from './doctor-card';
import DropdownItem from './dropdown-item';
import DropdownMenu from './dropdown-menu';
import Error from './error';
import EvaluateDialog from './evaluate-dialog';
import Field from './field';
import FileIcon from './file-icon';
import GoodsAction from './goods-action';
import GoodsActionButton from './goods-action-button';
import GoodsActionIcon from './goods-action-icon';
import Grid from './grid';
import GridItem from './grid-item';
import Guide from './guide';
import Icon from './icon';
import Image from './image';
import ImageDialog from './image-dialog';
import ImagePreview from './image-preview';
import IndexAnchor from './index-anchor';
import IndexBar from './index-bar';
import Info from './info';
import Lazyload from './lazyload';
import List from './list';
import Loading from './loading';
import Locale from './locale';
import NavBar from './nav-bar';
import Nodata from './nodata';
import NoticeBar from './notice-bar';
import Notify from './notify';
import NumberKeyboard from './number-keyboard';
import OtherCard from './other-card';
import Overlay from './overlay';
import Pagination from './pagination';
import Panel from './panel';
import PasswordInput from './password-input';
import PatientCard from './patient-card';
import PersonCard from './person-card';
import PiccodeDialog from './piccode-dialog';
import Picker from './picker';
import Popup from './popup';
import Progress from './progress';
import PullRefresh from './pull-refresh';
import Radio from './radio';
import RadioGroup from './radio-group';
import Rate from './rate';
import Result from './result';
import Row from './row';
import Schedule from './schedule';
import Search from './search';
import Sidebar from './sidebar';
import SidebarItem from './sidebar-item';
import Skeleton from './skeleton';
import Sku from './sku';
import Slider from './slider';
import Status from './status';
import Step from './step';
import Stepper from './stepper';
import Steps from './steps';
import Sticky from './sticky';
import SubmitBar from './submit-bar';
import Swipe from './swipe';
import SwipeCell from './swipe-cell';
import SwipeItem from './swipe-item';
import Switch from './switch';
import Tab from './tab';
import Tabbar from './tabbar';
import TabbarItem from './tabbar-item';
import Tabs from './tabs';
import Tag from './tag';
import TextBox from './text-box';
import TextDialog from './text-dialog';
import Toast from './toast';
import TreeSelect from './tree-select';
import Typography from './typography';
import Uploader from './uploader';
var version = '1.0.44';
var components = [ActionSheet, AddressEdit, AddressList, Area, Avatar, BankCard, BasicLayout, Body, Button, Call, Card, CascadeSelect, Cell, CellGroup, Checkbox, CheckboxGroup, Circle, Col, Collapse, CollapseItem, Color, ContactCard, ContactEdit, ContactList, CountDown, Coupon, CouponCell, CouponList, DatetimePicker, Dialog, Divider, DoctorCard, DropdownItem, DropdownMenu, Error, EvaluateDialog, Field, FileIcon, GoodsAction, GoodsActionButton, GoodsActionIcon, Grid, GridItem, Guide, Icon, Image, ImageDialog, ImagePreview, IndexAnchor, IndexBar, Info, List, Loading, NavBar, Nodata, NoticeBar, Notify, NumberKeyboard, OtherCard, Overlay, Pagination, Panel, PasswordInput, PatientCard, PersonCard, PiccodeDialog, Picker, Popup, Progress, PullRefresh, Radio, RadioGroup, Rate, Result, Row, Schedule, Search, Sidebar, SidebarItem, Skeleton, Sku, Slider, Status, Step, Stepper, Steps, Sticky, SubmitBar, Swipe, SwipeCell, SwipeItem, Switch, Tab, Tabbar, TabbarItem, Tabs, Tag, TextBox, TextDialog, Toast, TreeSelect, Typography, Uploader];

var install = function install(Vue) {
  components.forEach(function (Component) {
    Vue.use(Component);
  });
};
/* istanbul ignore if */


if (typeof window !== 'undefined' && window.Vue) {
  install(window.Vue);
}

export { install, version, ActionSheet, AddressEdit, AddressList, Area, Avatar, BankCard, BasicLayout, Body, Button, Call, Card, CascadeSelect, Cell, CellGroup, Checkbox, CheckboxGroup, Circle, Col, Collapse, CollapseItem, Color, ContactCard, ContactEdit, ContactList, CountDown, Coupon, CouponCell, CouponList, DatetimePicker, Dialog, Divider, DoctorCard, DropdownItem, DropdownMenu, Error, EvaluateDialog, Field, FileIcon, GoodsAction, GoodsActionButton, GoodsActionIcon, Grid, GridItem, Guide, Icon, Image, ImageDialog, ImagePreview, IndexAnchor, IndexBar, Info, Lazyload, List, Loading, Locale, NavBar, Nodata, NoticeBar, Notify, NumberKeyboard, OtherCard, Overlay, Pagination, Panel, PasswordInput, PatientCard, PersonCard, PiccodeDialog, Picker, Popup, Progress, PullRefresh, Radio, RadioGroup, Rate, Result, Row, Schedule, Search, Sidebar, SidebarItem, Skeleton, Sku, Slider, Status, Step, Stepper, Steps, Sticky, SubmitBar, Swipe, SwipeCell, SwipeItem, Switch, Tab, Tabbar, TabbarItem, Tabs, Tag, TextBox, TextDialog, Toast, TreeSelect, Typography, Uploader };
export default {
  install: install,
  version: version
};
