// vant-green@1.0.44/es/sidebar-item/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { createNamespace, addUnit } from '../utils';
import { ChildrenMixin } from '../mixins/relation';
import { route, routeProps } from '../utils/router';
import Info from '../info';
import Icon from '../icon';

var _createNamespace = createNamespace('sidebar-item'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

export default createComponent({
  mixins: [ChildrenMixin('vanSidebar')],
  props: _extends({}, routeProps, {
    dot: Boolean,
    info: [Number, String],
    title: String,
    disabled: Boolean,
    add: String,
    icon: String,
    iconSize: [String, Number]
  }),
  computed: {
    select: function select() {
      return this.index === +this.parent.activeKey;
    }
  },
  methods: {
    onClick: function onClick() {
      if (this.disabled) {
        return;
      }

      this.$emit('click', this.index);
      this.parent.$emit('input', this.index);
      this.parent.$emit('change', this.index);
      route(this.$router, this);
    }
  },
  render: function render() {
    var h = arguments[0];
    var contentStyle = {
      fontSize: addUnit(this.iconSize)
    };
    return h("a", {
      "class": bem({
        select: this.select,
        disabled: this.disabled,
        icon: this.icon
      }),
      "on": {
        "click": this.onClick
      }
    }, [h("div", {
      "class": bem('text')
    }, [this.slots('content'), this.icon && h("div", {
      "class": bem('icon-content')
    }, [h(Icon, {
      "attrs": {
        "name": this.icon
      },
      "style": contentStyle
    })]), h("div", {
      "class": bem('title')
    }, [this.title]), h(Info, {
      "attrs": {
        "dot": this.dot,
        "info": this.info
      },
      "class": bem('info')
    })]), !!this.add && h("div", {
      "class": bem('add')
    }, [this.add])]);
  }
});
