// vant-green@1.0.44/es/uploader/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { createNamespace, addUnit, noop } from '../utils';
import { toArray, readFile as _readFile, isOversize, isImageFile, random } from './utils';
import { BORDER_BOTTOM } from '../utils/constant';
import Icon from '../icon';
import Image from '../image';
import Loading from '../loading';
import ImagePreview from '../image-preview';

var _createNamespace = createNamespace('uploader'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

export default createComponent({
  inheritAttrs: false,
  model: {
    prop: 'fileList'
  },
  props: {
    disabled: Boolean,
    uploadText: String,
    afterRead: Function,
    beforeRead: Function,
    beforeDelete: Function,
    previewSize: [Number, String],
    name: {
      type: [Number, String],
      default: ''
    },
    accept: {
      type: String,
      default: 'image/*'
    },
    fileList: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    maxSize: {
      type: Number,
      default: Number.MAX_VALUE
    },
    maxCount: {
      type: Number,
      default: Number.MAX_VALUE
    },
    deletable: {
      type: Boolean,
      default: true
    },
    previewImage: {
      type: Boolean,
      default: true
    },
    previewFullImage: {
      type: Boolean,
      default: true
    },
    imageFit: {
      type: String,
      default: 'cover'
    },
    resultType: {
      type: String,
      default: 'dataUrl'
    },
    async: Boolean,
    progress: Boolean,
    progressInterval: {
      type: Number,
      default: 1000
    },
    title: String,
    description: String,
    tip: String
  },
  computed: {
    previewSizeWithUnit: function previewSizeWithUnit() {
      return addUnit(this.previewSize);
    }
  },
  methods: {
    getDetail: function getDetail(index) {
      if (index === void 0) {
        index = this.fileList.length;
      }

      return {
        name: this.name,
        index: index
      };
    },
    onChange: function onChange(event) {
      var _this = this;

      var files = event.target.files;

      if (this.disabled || !files.length) {
        return;
      }

      files = files.length === 1 ? files[0] : [].slice.call(files);

      if (this.beforeRead) {
        var response = this.beforeRead([].concat(files), this.getDetail());

        if (!response) {
          this.resetInput();
          return;
        }

        if (response.then) {
          response.then(function () {
            _this.readFile(files);
          }).catch(this.resetInput);
          return;
        }
      }

      this.readFile(files);
    },
    readFile: function readFile(files) {
      var _this2 = this;

      var oversize = isOversize(files, this.maxSize);

      if (Array.isArray(files)) {
        var maxCount = this.maxCount - this.fileList.length;

        if (files.length > maxCount) {
          files = files.slice(0, maxCount);
        }

        Promise.all(files.map(function (file) {
          return _readFile(file, _this2.resultType);
        })).then(function (contents) {
          var fileList = files.map(function (file, index) {
            var result = {
              file: file
            };

            if (contents[index]) {
              result.content = contents[index];
            }

            return result;
          });

          _this2.onAfterRead(fileList, oversize);
        });
      } else {
        _readFile(files, this.resultType).then(function (content) {
          var result = {
            file: files
          };

          if (content) {
            result.content = content;
          }

          _this2.onAfterRead(result, oversize);
        });
      }
    },
    onAfterRead: function onAfterRead(files, oversize) {
      if (oversize) {
        this.$emit('oversize', files, this.getDetail());
        return;
      }

      this.resetInput();
      files = toArray(files);

      if (!this.async) {
        files.forEach(function (element) {
          element.status = 'loading';
        });
      }

      this.$emit('input', [].concat(this.fileList, files));
      this.afterRead && this.afterRead(files, this.getDetail());
    },
    onDelete: function onDelete(file, index) {
      var _this3 = this;

      if (this.beforeDelete) {
        var response = this.beforeDelete(file, this.getDetail(index));

        if (!response) {
          return;
        }

        if (response.then) {
          response.then(function () {
            _this3.deleteFile(file, index);
          }).catch(noop);
          return;
        }
      }

      this.deleteFile(file, index);
    },
    deleteFile: function deleteFile(file, index) {
      var fileList = this.fileList.slice(0);
      fileList.splice(index, 1);
      this.$emit('input', fileList);
      this.$emit('delete', file, this.getDetail(index));
    },
    resetInput: function resetInput() {
      /* istanbul ignore else */
      if (this.$refs.input) {
        this.$refs.input.value = '';
      }
    },
    onPreviewImage: function onPreviewImage(item) {
      var _this4 = this;

      if (!this.previewFullImage) {
        return;
      }

      var imageFiles = this.fileList.filter(function (item) {
        return isImageFile(item);
      });
      var imageContents = imageFiles.map(function (item) {
        return item.content || item.url;
      });
      this.imagePreview = ImagePreview({
        images: imageContents,
        closeOnPopstate: true,
        startPosition: imageFiles.indexOf(item),
        onClose: function onClose() {
          _this4.$emit('close-preview');
        }
      });
    },
    closeImagePreview: function closeImagePreview() {
      if (this.imagePreview) {
        this.imagePreview.close();
      }
    },
    getRate: function getRate(item) {
      var randomUp = random(70, 90);
      var randomNumber = random(2, 10);

      if (!item.hasOwnProperty('rate')) {
        this.$set(item, 'rate', randomNumber);
      }

      var timer = setInterval(function () {
        if (item.rate < randomUp) {
          item.rate += randomNumber;
        } else {
          clearInterval(timer);
        }
      }, this.progressInterval);
    },
    getStatus: function getStatus(item) {
      var _this5 = this;

      var h = this.$createElement;

      switch (item.status) {
        case 'loading':
          return h("div", {
            "class": bem('file-status')
          }, [h(Loading, {
            "attrs": {
              "type": "spinner",
              "size": "20"
            },
            "class": bem('file-status-icon')
          }), this.progress && this.getRate(item), item.rate && h("p", [item.rate, "%"])]);

        case 'fail':
          return h("div", {
            "class": bem('file-status'),
            "on": {
              "click": function click() {
                item.hasOwnProperty('rate') && delete item.rate;
                item.status = 'loading';
                _this5.afterRead && _this5.afterRead([item], 'reupload');
              }
            }
          }, [h("p", ["\u4E0A\u4F20\u5931\u8D25"]), h("p", ["\u70B9\u51FB\u91CD\u8BD5"])]);

        default:
          return h("div", [h(Icon, {
            "class": bem('file-icon'),
            "attrs": {
              "name": "description"
            }
          }), h("div", {
            "class": [bem('file-name'), 'van-ellipsis']
          }, [item.file ? item.file.name : item.url])]);
      }
    },
    genPreviewItem: function genPreviewItem(item, index) {
      var _this6 = this;

      var h = this.$createElement;
      var DeleteIcon = this.deletable && item.status !== 'loading' && h("span", {
        "class": bem('preview-delete'),
        "on": {
          "click": function click(event) {
            event.stopPropagation();

            _this6.onDelete(item, index);
          }
        }
      }, [h(Icon, {
        "attrs": {
          "name": "cross"
        }
      })]);
      var Preview = isImageFile(item) && !item.status ? h(Image, {
        "attrs": {
          "fit": this.imageFit,
          "src": item.content || item.url,
          "width": this.previewSize,
          "height": this.previewSize
        },
        "class": bem('preview-image'),
        "on": {
          "click": function click() {
            _this6.onPreviewImage(item);
          }
        }
      }) : h("div", {
        "class": bem('file', {
          fail: item.status === 'fail'
        }),
        "style": {
          width: this.previewSizeWithUnit,
          height: this.previewSizeWithUnit
        }
      }, [this.getStatus(item)]);
      return h("div", {
        "class": bem('preview'),
        "on": {
          "click": function click() {
            _this6.$emit('click-preview', item, _this6.getDetail(index));
          }
        }
      }, [Preview, DeleteIcon]);
    },
    genPreviewList: function genPreviewList() {
      if (this.previewImage) {
        return this.fileList.map(this.genPreviewItem);
      }
    },
    genUpload: function genUpload() {
      var h = this.$createElement;

      if (this.fileList.length >= this.maxCount) {
        return;
      }

      var slot = this.slots();
      var Input = h("input", {
        "attrs": _extends({}, this.$attrs, {
          "type": "file",
          "accept": this.accept,
          "disabled": this.disabled
        }),
        "ref": "input",
        "class": bem('input'),
        "on": {
          "change": this.onChange
        }
      });

      if (slot) {
        return h("div", {
          "class": bem('input-wrapper')
        }, [slot, Input]);
      }

      var style;

      if (this.previewSize) {
        var size = this.previewSizeWithUnit;
        style = {
          width: size,
          height: size
        };
      }

      return h("div", {
        "class": bem('upload'),
        "style": style
      }, [h("i", {
        "class": bem('plus')
      }), this.uploadText && h("span", {
        "class": bem('upload-text')
      }, [this.uploadText]), Input]);
    },
    getTitle: function getTitle() {
      var h = this.$createElement;
      return this.title && h("div", {
        "class": [bem('title'), BORDER_BOTTOM]
      }, [h("p", [this.title, h("span", {
        "class": bem('title-description')
      }, [this.description])]), this.maxCount && this.maxCount < Number.MAX_VALUE && h("p", {
        "class": bem('title-right')
      }, [h("span", {
        "class": bem([this.fileList.filter(function (obj) {
          return !obj.status;
        }).length >= this.maxCount ? 'max-count' : this.fileList.filter(function (obj) {
          return !obj.status;
        }).length > 0 ? 'count' : ''])
      }, [this.fileList.filter(function (obj) {
        return !obj.status;
      }).length]), "/", this.maxCount])]);
    },
    getTip: function getTip() {
      var h = this.$createElement;
      return this.tip && h("div", {
        "class": bem('tip')
      }, [this.tip]);
    }
  },
  render: function render() {
    var h = arguments[0];
    return h("div", {
      "class": bem()
    }, [this.getTitle(), h("div", {
      "class": bem('wrapper')
    }, [this.genPreviewList(), this.genUpload()]), this.getTip()]);
  }
});
