// vant-green@1.0.44/es/address-list/Item.js
import _mergeJSXProps from "@vue/babel-helper-vue-jsx-merge-props";
import { createNamespace } from '../utils';
import { emit, inherit } from '../utils/functional';
import Icon from '../icon';
import Cell from '../cell';
import Radio from '../radio';
import Tag from '../tag'; // Types

var _createNamespace = createNamespace('address-item'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

function AddressItem(h, props, slots, ctx) {
  var disabled = props.disabled,
      switchable = props.switchable;

  function onClick() {
    if (switchable) {
      emit(ctx, 'select');
    }

    emit(ctx, 'click');
  }

  var genRightIcon = function genRightIcon() {
    return h(Icon, {
      "attrs": {
        "name": "edit"
      },
      "class": bem('edit'),
      "on": {
        "click": function click(event) {
          event.stopPropagation();
          emit(ctx, 'edit');
          emit(ctx, 'click');
        }
      }
    });
  };

  var genContent = function genContent() {
    var data = props.data;
    var Info = [h("div", {
      "class": bem('title')
    }, [h("span", {
      "class": bem('name')
    }, [data.name]), data.tel, data.default && h(Tag, {
      "attrs": {
        "type": "success"
      },
      "class": bem('tag')
    }, ["\u9ED8\u8BA4"])]), h("div", {
      "class": [bem('address'), 'van-ellipsis']
    }, [data.address])];
    return switchable && !disabled ? h(Radio, {
      "attrs": {
        "name": data.id,
        "iconSize": 22
      },
      "class": bem('radio')
    }, [Info]) : Info;
  };

  return h(Cell, _mergeJSXProps([{
    "class": bem({
      disabled: disabled
    }),
    "attrs": {
      "valueClass": bem('value'),
      "clickable": switchable && !disabled
    },
    "scopedSlots": {
      default: genContent,
      'right-icon': genRightIcon
    },
    "on": {
      "click": onClick
    }
  }, inherit(ctx)]));
}

AddressItem.props = {
  data: Object,
  disabled: Boolean,
  switchable: Boolean
};
export default createComponent(AddressItem);
