// vant-green@1.0.44/lib/person-card/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _utils = require("../utils");

var _avatar = _interopRequireDefault(require("../avatar"));

var _row = _interopRequireDefault(require("../row"));

var _tag = _interopRequireDefault(require("../tag"));

var _createNamespace = (0, _utils.createNamespace)('person-card'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

var _default = createComponent({
  props: {
    avatarFigure: {
      type: [String, Number],
      default: 1
    },
    avatarSize: {
      type: [Number, String],
      default: 50
    },
    avatarRadius: [Number, String],
    title: String,
    subTitle: String,
    tag: String,
    tagType: String,
    corner: String,
    content: [String, Array]
  },
  render: function render() {
    var _this = this;

    var h = arguments[0];
    var slots = this.slots;
    var genAvatar = h("div", {
      "class": bem('avatar')
    }, [slots('avatar') || h(_avatar.default, {
      "attrs": {
        "figure": this.avatarFigure,
        "size": this.avatarSize,
        "radius": this.avatarRadius
      }
    })]);
    var genBase = h(_row.default, {
      "attrs": {
        "type": "flex",
        "align": "center",
        "justify": "space-between"
      },
      "class": bem('base')
    }, [h(_row.default, {
      "attrs": {
        "type": "flex",
        "align": "center"
      }
    }, [slots('title') || h("div", [(0, _utils.isDef)(this.title) && h("span", {
      "class": bem('title')
    }, [this.title]), (0, _utils.isDef)(this.subTitle) && h("span", {
      "class": bem('sub-title')
    }, [this.subTitle])]), this.slots('tag') || this.tag && h(_tag.default, {
      "attrs": {
        "type": this.tagType,
        "round": true
      }
    }, [this.tag])]), h("div", [this.slots('corner') || h("span", {
      "class": bem('corner')
    }, [this.corner])])]);
    var genContent = h("div", {
      "class": bem('content')
    }, [slots() || typeof this.content === 'string' ? h("span", {
      "class": "van-multi-ellipsis--l1"
    }, [this.content]) : this.content.map(function (item, index) {
      return h("span", {
        "class": bem('content-item')
      }, [item, h("i", {
        "directives": [{
          name: "show",
          value: index !== _this.content.length - 1
        }],
        "class": bem('gap')
      })]);
    })]);
    return h("div", {
      "attrs": {
        "role": "person-card"
      },
      "class": bem()
    }, [genAvatar, h("div", {
      "class": bem('wrapper')
    }, [genBase, genContent])]);
  }
});

exports.default = _default;
