// vant-green@1.0.44/lib/submit-bar/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _babelHelperVueJsxMergeProps = _interopRequireDefault(require("@vue/babel-helper-vue-jsx-merge-props"));

var _utils = require("../utils");

var _functional = require("../utils/functional");

var _constant = require("../utils/constant");

var _button = _interopRequireDefault(require("../button"));

var _icon = _interopRequireDefault(require("../icon"));

var _createNamespace = (0, _utils.createNamespace)('submit-bar'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1],
    t = _createNamespace[2];

function SubmitBar(h, props, slots, ctx) {
  var tip = props.tip,
      price = props.price,
      tipIcon = props.tipIcon;

  function Text() {
    if (typeof price === 'number') {
      var priceText = "" + (price / 100).toFixed(props.decimalLength);
      return h("div", {
        "class": bem('text')
      }, [h("span", {
        "class": bem('label')
      }, [props.label || t('label')]), h("span", {
        "class": bem('price')
      }, [props.currency !== '元' && h("i", {
        "class": bem('currency')
      }, [props.currency]), h("span", {
        "class": bem('price-text')
      }, [priceText]), props.currency === '元' && h("i", {
        "class": bem('currency')
      }, [props.currency])]), h("span", {
        "class": bem('suffix-label')
      }, [slots.suffixLabel && slots.suffixLabel() || props.suffixLabel])]);
    }
  }

  function Tip() {
    if (slots.tip || tip) {
      return h("div", {
        "class": bem('tip')
      }, [tipIcon && h(_icon.default, {
        "class": bem('tip-icon'),
        "attrs": {
          "name": tipIcon
        }
      }), tip && h("span", {
        "class": bem('tip-text')
      }, [tip]), slots.tip && slots.tip()]);
    }
  }

  return h("div", {
    "class": bem('wrapper'),
    "style": {
      height: (0, _utils.addUnit)(props.height)
    }
  }, [h("div", (0, _babelHelperVueJsxMergeProps.default)([{
    "class": [bem({
      'safe-area-inset-bottom': props.safeAreaInsetBottom
    }), props.border ? _constant.BORDER_TOP : '']
  }, (0, _functional.inherit)(ctx)]), [slots.top && slots.top(), Tip(), h("div", {
    "class": bem('bar')
  }, [slots.default && slots.default(), !slots.default && slots.left && slots.left(), !slots.default && Text(), !slots.default && h(_button.default, {
    "class": bem('button'),
    "attrs": {
      "type": props.buttonType,
      "loading": props.loading,
      "disabled": props.disabled,
      "text": props.loading ? '' : props.buttonText
    },
    "on": {
      "click": function click() {
        (0, _functional.emit)(ctx, 'submit');
      }
    }
  })])])]);
}

SubmitBar.props = {
  tip: String,
  label: String,
  price: Number,
  tipIcon: String,
  loading: Boolean,
  disabled: Boolean,
  buttonText: String,
  suffixLabel: String,
  safeAreaInsetBottom: {
    type: Boolean,
    default: true
  },
  decimalLength: {
    type: Number,
    default: 2
  },
  currency: {
    type: String,
    default: '元'
  },
  buttonType: {
    type: String,
    default: 'warning'
  },
  border: {
    type: Boolean,
    default: true
  },
  height: [Number, String]
};

var _default = createComponent(SubmitBar);

exports.default = _default;
